### PR TITLE
docs: document errors

### PR DESCRIPTION
**Pull request by keroro520**

### CODE REVIEW

File: error/src/util.rs [hunk 1]
```diff
@@ -23,7 +24,27 @@ macro_rules! assert_error_eq {
     }
 }
 
-/// TODO(doc): @keroro520
+/// A macro to implement conversion from source type to target type with an implicit error kind.
+///
+/// ## Examples
+///
+/// ```rust
+/// impl_error_conversion_with_kind!(SourceType, error_kind, TargetType)
```
Comment (yangby-cryptape) [inline error/src/util.rs:32]:
```suggestion
/// ```ignore
/// impl_error_conversion_with_kind!(SourceType, error_kind, TargetType);
```

I didn't run the tests, but I guess this code couldn't pass `cargo run`.

File: error/src/util.rs [hunk 2]
```diff
@@ -23,7 +24,27 @@ macro_rules! assert_error_eq {
...
+/// ```
+///
+/// the expanded code:
+///
+/// ```rust
```
Comment (yangby-cryptape) [inline error/src/util.rs:37]:
```suggestion
/// ```ignore
```

File: error/src/util.rs [hunk 3]
```diff
@@ -35,7 +56,26 @@ macro_rules! impl_error_conversion_with_kind {
...
+///
+/// ## Examples
+///
+/// ```rust
+/// impl_error_conversion_with_adaptor!(SourceType, AdaptorType, TargetType)
+/// ```
+///
+/// the expanded code:
+///
+/// ```rust
```
Comment (yangby-cryptape) [inline error/src/util.rs:69]:
```suggestion
/// ```ignore
```

File: error/src/util.rs [hunk 4]
```diff
@@ -35,7 +56,26 @@ macro_rules! impl_error_conversion_with_kind {
     };
 }
 
-/// TODO(doc): @keroro520
+/// A macro to implement conversion from source type to target type based on an implicit middle adaptor.
+///
+/// ## Examples
+///
+/// ```rust
+/// impl_error_conversion_with_adaptor!(SourceType, AdaptorType, TargetType)
```
Comment (yangby-cryptape) [inline error/src/util.rs:64]:
```suggestion
/// ```ignore
/// impl_error_conversion_with_adaptor!(SourceType, AdaptorType, TargetType);
```

File: error/Cargo.toml [hunk 5]
```diff
@@ -4,7 +4,7 @@ version = "0.39.0-pre"
 license = "MIT"
 authors = ["Nervos Core Dev <dev@nervos.org>"]
 edition = "2018"
-description = "TODO(doc): @keroro520 crate description"
+description = "underlying error types used over ckb crates"
```
Comment (yangby-cryptape) [inline error/Cargo.toml:7]:
```suggestion
description = "Underlying error types used over ckb crates"
```

File: util/types/src/core/error.rs [hunk 6]
```diff
@@ -1,36 +1,35 @@
...
 
-    // TODO: This error should be move into HeaderError or TransactionError
-    /// TODO(doc): @keroro520
+    /// There is a dependency header that is unknown to the canonical chain.
     #[error("InvalidHeader({0})")]
     InvalidHeader(Byte32),
 
-    // TODO: This error should be move into HeaderError or TransactionError
-    /// TODO(doc): @keroro520
+    /// There is a dependancy header that is immature yet.
```
Comment (yangby-cryptape) [inline util/types/src/core/error.rs:32]:
```suggestion
    /// There is a dependency header that is immature yet.
```

File: util/dao/src/lib.rs [hunk 7]
```diff
@@ -97,7 +98,10 @@ impl<'a, CS: ChainStore<'a>> DaoCalculator<'a, CS, DataLoaderWrapper<'a, CS>> {
         Ok(primary_block_reward.safe_add(secondary_block_reward)?)
     }
 
-    /// TODO(doc): @keroro520
+    /// Calculates the new dao field after packaging these transactions. It returns the dao field in [`Byte32`] format. Please see [`extract_dao_data`] if you intend to see the detailed content.
```
Comment (yangby-cryptape) [inline util/dao/src/lib.rs:101]:
```suggestion
    /// Calculates the new dao field after packaging these transactions. It returns the dao field in [`Byte32`] format.
    /// Please see [`extract_dao_data`] if you intend to see the detailed content.
```

Too long.

**Review by yangby-cryptape — CHANGES_REQUESTED** [review]

**Comment by keroro520** [comment]
@yangby-cryptape Thanks for your comments. Most of the suggestions applied, except for the long line.

There are so many long comments in this PR and original code; I would like to ignore this issue for the moment and unify them via [rustfmt comment_width](https://rust-lang.github.io/rustfmt/?version=master&search=#comment_width) later.

**Review by yangby-cryptape — APPROVED** [review]

**Review by doitian — APPROVED** [review]

**Comment by doitian** [comment]
bors r=doitian,yangby-cryptape

**Pull request merged**